### PR TITLE
[CI] Fix Lightning versions in CPL actions

### PR DESCRIPTION
**Pull request by maliasadi**
- [X] Update the CPL actions to properly install `pennylane-lightning` against the master branch (latest) or  the most recent release (stable). 
- [X] Update the Lightning-Kokkos support based on the repo changes in the master branch (`v0.30.0rc0`).

### CODE REVIEW

**Comment by dime10** [comment]
What are all the runtime changes?

File: .github/workflows/check-pl-compat.yaml [hunk 1]
```diff
@@ -102,6 +102,20 @@ jobs:
...
+        repository: PennyLaneAI/pennylane-lightning
+        path: pennylane-lightning
+    - if: ${{ inputs.lightning == 'latest' }}
+      run: |
+        pip install --upgrade ./pennylane-lightning
+
+    - name: Install PennyLane-Lightning (stable)
+      if: ${{ inputs.lightning == 'stable' }}
+      run: |
+        pip install --upgrade pennylane-lightning
```
Comment (dime10) [inline .github/workflows/check-pl-compat.yaml:118]: Thanks for catching this!

**Review by dime10 — APPROVED** [review]

File: .github/workflows/check-pl-compat.yaml (hunk 1 above)
Comment (dime10) [reply .github/workflows/check-pl-compat.yaml:118]: Do you think it's issue that we install lightning before PennyLane? The version is not going to be overwritten during the PL installation right? (unless the versions conflict with PL's dep requirements 🤔)

**Comment by maliasadi** [comment]
> What are all the runtime changes?

These updates are related to the recent changes in the master branch of Lightning-Kokkos.

File: .github/workflows/check-pl-compat.yaml (hunk 1 above)
Comment (maliasadi) [reply .github/workflows/check-pl-compat.yaml:118]: There shouldn't be any issue as PL only checks if the installed pennylane-lightning version is satisfying the minimum (v0.28.0) if not it'll installs the stable release.

**Pull request merged**